### PR TITLE
`transaction` - add skeleton loader

### LINKED ISSUE #690: `transaction` - add skeleton loader
#### What UI/UX improvements would you like to be implemented: 
- Display skeleton loader when:
  - first load
  - pull to refresh
- Background update should not trigger skeleton loader

#### Why is this needed:
- To inform user that the data is loading

#### Which component/screens does this concern?:
Transaction Screen

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- Add skeleton loader to inform user that data is loading
- Fix "ghost pulling" that was caused by background update triggerring refresh action

#### Which issue(s) does this PR fixes?:

Fixes #690

#### Additional comments?:
Tested on iPhone 12 and pixel 3a emulator

### CODE REVIEW

File: app/store/transaction.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { VMTransaction } from '../screens/AppNavigator/screens/Transactions/screens/stateProcessor'
+
+type LoadingState = 'idle' | 'loading' | 'loadingMore' | 'success' | 'background' | 'error'
+
+export interface TransactionState {
```
Comment (thedoublejay) [inline app/store/transaction.ts:6]: Why did you add it on the global state? It's not used anywhere else except Transaction Page

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: app/store/transaction.ts (hunk 1 above)
Comment (kyleleow) [reply app/store/transaction.ts:6]: My intention is to move the side effects that is not related to the rendering logic outside from the component. 

But I couldn't achieve what I wanted to do when I tried to make async call using redux's createAsyncThunk(). Hence I settled with this solution.

Will there be any scenario or feature in the future that we might need to access transaction data anywhere else? If there is none then I'll move the state out from redux.

Comment (thedoublejay) [reply app/store/transaction.ts:6]: Nope, we will only use it on this page. Yes please move it out of the global store

Comment (kyleleow) [reply app/store/transaction.ts]: Updated

File: cypress/integration/functional/wallet/transaction/list.spec.ts [hunk 2]
```diff
@@ -1,10 +1,29 @@
 context('Wallet - Transaction - List', () => {
   describe('wallet has 2 transactions in history', function () {
     before(() => {
+      cy.visit('/')
```
Comment (thedoublejay) [inline cypress/integration/functional/wallet/transaction/list.spec.ts:4]: @kyleleow this one is already done inside createEmptyWallet

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: cypress/integration/functional/wallet/transaction/list.spec.ts (hunk 2 above)
Comment (kyleleow) [reply cypress/integration/functional/wallet/transaction/list.spec.ts]: updated

**Review by thedoublejay — APPROVED** [review]
